server/src/repositories/test_repository.ts

```diff
@@ -0,0 +1,30 @@
+import { inject, injectable } from "inversify";
+import { Pool } from "pg";
+import { TestResults } from "../entities/test";
+
+@injectable()
+export class TestRepository {
+    constructor(@inject("DBConnectionPool") private readonly pool: Pool) {}
+
+    async addTestResult(testResults: TestResults) {
+        const client = await this.pool.connect();
+
+        const queryString = `INSERT INTO test_results 
+                            (patient_id, 
+                             test_result, 
+                             type_of_test, 
+                             date_of_test,
+                             address_id)
+                             VALUES ($1, $2, $3, $4, $5)`;
```

Comment:
I would update these in the schema so that it maps to the domain model entities better

```suggestion
        const queryString = `INSERT INTO test_results 
                            (patient_id, 
                             result, 
                             test_type, 
                             test_date,
                             address_id)
                             VALUES ($1, $2, $3, $4, $5)`;
```